testsuite/classlibrary/TestPattern.sc

```diff
@@ -155,6 +155,21 @@ TestPattern : UnitTest {
 		this.assert(x.sum { |x| x.delta } == 4.5, "Psync with maxdur = quant should end no earlier than after dur");
 	}
 
+	test_Pfset_nilStream {
+		var x = 0, y = 0, oe = (), ie = (), clup = EventStreamCleanup.new;
```

Comment:
I'm being picky, but for future maintenance, I'd suggest more descriptive variable names. `x` and `y` are fine, but there's no runtime cost to say `outEvent` rather than `oe` or `cleanup` for `clup`.

Also the `oe` initializer is redundant because `oe` is immediately assigned a different value. (If the initializer were simply a numeric constant, there would be no real impact -- but `()` really translates to `Event.new` and there is a small runtime cost to constructing a new object and immediately discarding it. Here, SC differs from other languages that warn about uninitialized variables.)

So I'd say `var x, y, outEvent, inEvent = (), cleanup = EventStreamCleanup.new;` (no need really to initialize x and y either -- if you need to know the init func didn't run, `x.isNil` is sufficient).

Reply:
> Also the oe initializer is redundant because oe is immediately assigned a different value. 

It's not redundant in this case because were are testing Pfset's return value as well (with isNil in the 3rd assert). If Pfset has a bug and doesn't change oe (outEvent) back to nil, that would be a bug in Pfset we'd want to detect. But we would not be able to do that if we were to let oe (outEvent) have the initial default value nil by not initializing it to something else than nil before calling Pfset. The rest of your suggestions, I agree with and I'm going to push a change accordingly.